src/cluster/cluster.cc

```diff
@@ -157,7 +159,34 @@ Status Cluster::SetSlot(int slot, const std::string &node_id, int64_t new_versio
 
   return Status::OK();
 }
+// Get all local IP addresses.
+std::vector<std::string> getLocalIPAddresses() {
+  std::vector<std::string> ipAddresses;
+
+  struct ifaddrs *ifAddrStruct = NULL;
+  struct ifaddrs *ifa = NULL;
+  void *tmpAddrPtr = NULL;
+
+  getifaddrs(&ifAddrStruct);
 
+  for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next) {
+    if (!ifa->ifa_addr) {
+      continue;
+    }
+    if (ifa->ifa_addr->sa_family == AF_INET) {
+      // check it is IP4 and not a loopback address
+      tmpAddrPtr = &((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+      char addressBuffer[INET_ADDRSTRLEN];
+      inet_ntop(AF_INET, tmpAddrPtr, addressBuffer, INET_ADDRSTRLEN);
+      std::string ipAddress(addressBuffer);
+      ipAddresses.push_back(move(ipAddress));
```

Comment:
I would add `std::` prefix to the `move` function.

Reply:
It looks unnecessary to get IP addresses, we can do it once if finds the listening address was on 0.0.0.0 when starting.

Reply:
Can I understand that if the server listens to 0.0.0.0, any cluster node with any ip on the same port needs to be matched, or what?

Reply:
No actually, we can only match the IP addresses that the current node is listening. And 0.0.0.0 is a special case since it represents listening on all IP addresses in this node, we need to convert to the real IP(like 192.168.1.4) + loopback IP(127.0.0.1) before matching.

Reply:
So is there a problem with my above operation? If it is 0.0.0.0, get all local ip and cluster node ip for matching

Reply:
The implementation of `getLocalIPAddresses` looks good. My point is that we can use all IP addresses only if the listening address is `0.0.0.0`.

Reply:
Got it, thanks